logzio/resource_log_shipping_token.go

```diff
@@ -0,0 +1,196 @@
+package logzio
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/logzio/logzio_terraform_client/log_shipping_tokens"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	logShippingTokenId        = "id"
+	logShippingTokenName      = "name"
+	logShippingTokenEnabled   = "enabled"
+	logShippingTokenToken     = "token"
+	logShippingTokenUpdatedAt = "updated_at"
+	logShippingTokenUpdatedBy = "updated_by"
+	logShippingTokenCreatedAt = "created_at"
+	logShippingTokenCreatedBy = "created_by"
+	logShippingTokenTokenId   = "token_id"
+)
+
+func resourceLogShippingToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLogShippingTokenCreate,
+		Read:   resourceLogShippingTokenRead,
+		Update: resourceLogShippingTokenUpdate,
+		Delete: resourceLogShippingTokenDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			logShippingTokenTokenId: {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			logShippingTokenName: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			logShippingTokenEnabled: {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			logShippingTokenToken: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			logShippingTokenUpdatedAt: {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			logShippingTokenUpdatedBy: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			logShippingTokenCreatedAt: {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			logShippingTokenCreatedBy: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Second),
+			Update: schema.DefaultTimeout(5 * time.Second),
+			Delete: schema.DefaultTimeout(5 * time.Second),
+		},
+	}
+}
+
+// Creates a new log shipping token in logz.io
+func resourceLogShippingTokenCreate(d *schema.ResourceData, m interface{}) error {
+	createToken := log_shipping_tokens.CreateLogShippingToken{Name: d.Get(logShippingTokenName).(string)}
+	tokenLimits, err := logShippingTokenClient(m).GetLogShippingLimitsToken()
+	if err != nil {
+		return err
+	}
+
+	// Check if we exceeded the number of max allowed tokens
+	if tokenLimits.MaxAllowedTokens > tokenLimits.NumOfEnabledTokens {
```

Comment:
can you replace the order?
tokenLimits.NumOfEnabledTokens <= tokenLimits.MaxAllowedTokens 
Its more readable that way

Reply:
I think it should be < rather than <=. if it equal they can't create more.

Reply:
wasn't sure but if you say so i believe you
anyway my comment was on the order to make it more readable